extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResourceGeneratedContentForSchedule.java

```diff
@@ -0,0 +1,124 @@
+/*!
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2018 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.web.http.api.resources;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.enunciate.jaxrs.ResponseCode;
+import org.codehaus.enunciate.jaxrs.StatusCodes;
+import org.pentaho.platform.api.repository2.unified.webservices.RepositoryFileDto;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+/**
+ * This service provides methods for listing, creating, downloading, uploading, and removal of files.
+ *
+ * @author aaron
+ */
+@Path ( "/repo/files/generatedContentForSchedule" )
+public class FileResourceGeneratedContentForSchedule extends AbstractJaxRSResource {
```

Comment:
If I'm understanding correctly, that line in `pentahoServices.spring.xml` **must** be present, otherwise we lose access to the public-facing API of https://help.pentaho.com/Documentation/8.1/Developer_Center/REST_API/File_Management/0O0#section_143

Am I understanding it correctly ?

Reply:
Yes